Gems/ROS2/Code/CMakeLists.txt

```diff
@@ -161,6 +161,8 @@ if(PAL_TRAIT_BUILD_HOST_TOOLS)
             PUBLIC
                 Gem::${gem_name}.Editor.Static
                 Gem::Atom_Feature_Common.Static
+            PRIVATE
+                AZ::AzToolsFramework
```

Comment:
This is not needed.
The `ROS2.Editor` MODULE target gets the AzToolsFramework dependency via the `ROS2.Editor.Static` target above since it exposes `AzToolsFramework` as a public dependency